constants.php

```diff
@@ -1,6 +1,38 @@
 <?php
 
-namespace Fig\Http\Message;
+/**
+ * Defines constants for common HTTP request methods.
+ *
+ * Usage:
+ *
+ * <code>
+ * use Fig\Http\Message\Request;
+ *
+ * class RequestFactory
+ * {
+ *     public static function factory(
+ *         $uri = '/',
+ *         $method = Request\METHOD_GET,
+ *         $data = []
+ *     ) {
+ *     }
+ * }
+ * </code>
+ */
+namespace Fig\Http\Message\Request
```

Comment:
This namespace is kind of weird. Would be nice to write code like:

```php
use const Http\Request\Method\GET;
use const Http\Response\Status\NOT_FOUND;
```

Though I suppose some of these constants might be reserved words in PHP <7.

Reply:
what about `Fig\Http` as a root instead of just `Http`? We need a vendor prefix.

Reply:
Why is a vendor prefix necessary? @http-interop doesn't use one.

Reply:
I need to remind you the bylaw about this: https://www.php-fig.org/bylaws/psr-naming-conventions/

> Special lighweight utility packages MAY be produced alongside PSRs and interfaces and be maintained and updated after the PSR has been accepted. These MUST be under the vendor namespace `Fig`.

Reply:
> Why is a vendor prefix necessary?

oh come on Woody, really? Even if we didn't have the bylaw, it is not a great idea to claim a namespace as generic as `Http`... 😜

Reply:
> "Claim" a namespace? One package using a namespace does not prevent another package from using it.

Ok, "claim" is not entirely accurate, but that's what happens de-facto. Besides, it's asking for trouble!